tests/components/smappee/test_config_flow.py

```diff
@@ -10,7 +19,181 @@
+    """Test that the zeroconf confirmation form is served."""
+    with patch("pysmappee.api.SmappeeLocalApi.logon", return_value={}), patch(
+        "pysmappee.api.SmappeeLocalApi.load_advanced_config",
+        return_value=[{"key": "mdnsHostName", "value": "Smappee1006000212"}],
+    ), patch(
+        "pysmappee.api.SmappeeLocalApi.load_command_control_config", return_value=[]
+    ), patch(
+        "pysmappee.api.SmappeeLocalApi.load_instantaneous",
+        return_value=[{"key": "phase0ActivePower", "value": 0}],
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_ZEROCONF},
+            data={
+                "host": "1.2.3.4",
+                "port": 22,
+                CONF_HOSTNAME: "Smappee1006000212.local.",
+                "type": "_ssh._tcp.local.",
+                "name": "Smappee1006000212._ssh._tcp.local.",
+                "properties": {"_raw": {}},
+            },
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "zeroconf_confirm"
+        assert result["description_placeholders"] == {CONF_SERIALNUMBER: "1006000212"}
+
+
+async def test_show_zeroconf_connection_error_form(hass):
+    """Test that the zeroconf confirmation form is served."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data={
+            "host": "1.2.3.4",
+            "port": 22,
+            CONF_HOSTNAME: "Smappee1006000212.local.",
+            "type": "_ssh._tcp.local.",
+            "name": "Smappee1006000212._ssh._tcp.local.",
+            "properties": {"_raw": {}},
+        },
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "connection_error"
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 0
+
+
+async def test_connection_error(hass):
+    """Test we show user form on Smappee connection error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data={"host": "1.2.3.4"},
+    )
+
+    assert result["reason"] == "connection_error"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 0
+
+
+async def test_connection_error_empty_host(hass):
+    """Test we show user form on Smappee connection error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data={"host": None},
+    )
+
+    assert result["reason"] == "connection_error"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 0
+
+
+async def test_zeroconf_no_data(hass):
+    """Test we abort if zeroconf provides no data."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_ZEROCONF}
```

Comment:
This can never happen.

Reply:
What is the difference with this test https://github.com/home-assistant/core/blob/569caf9e402c22b11f5624a025b9529bc2ed26e3/tests/components/wled/test_config_flow.py#L121 ?

Reply:
That test originates from a time where the `discovery_info` was still marked as optional. Hence I added that for coverage. At this point, it isn't needed anymore (and thus WLED needs cleanup).